src/git/tag.go

```diff
@@ -94,3 +106,19 @@ func (s *RepoTagsGetter) Tags() ([]Tag, error) {
 
 	return tags, nil
 }
+
+// getCommitsMap retrieves the list of commits from HEAD and put it in a map in order to
+// filter out all the tags not belonging to the current branch.
+func getCommitsMap(commitGetter CommitsGetter) (map[string]bool, error) {
+	branchCommits, err := commitGetter.Commits(EmptyTreeID)
+	if err != nil {
+		return nil, fmt.Errorf("getting git commits since empty tree: %w", err)
+	}
+
+	commitMap := map[string]bool{}
+	for _, c := range branchCommits {
+		commitMap[c.Hash] = true
+	}
+
+	return commitMap, nil
+}
```

Comment:
This dangling (i.e. not tied to an object) function is a very good pointer that we should have an object doing this work.